docs/reference/search/vector-tile-api.asciidoc

```diff
@@ -0,0 +1,717 @@
+[[vector-tile-api]]
+=== Vector tile API
```

Comment:
I went with "Vector tile" as the API name. Let me know if you prefer something different.